cloud/deployment/deployment.go

```diff
@@ -274,105 +291,275 @@ func Create(label, workspaceID, description, clusterID, runtimeVersion, dagDeplo
-		return err
-	}
-
-	scheduler := astro.Scheduler{
-		AU:       schedulerAU,
-		Replicas: schedulerReplicas,
-	}
-
-	spec := astro.DeploymentCreateSpec{
-		Executor:  executor,
-		Scheduler: scheduler,
+	if !IsDeploymentStandard(deploymentType) {
+		clusterID, err = selectCluster(clusterID, c.Organization, corePlatformClient)
+		if err != nil {
+			return err
+		}
 	}
 
-	if dagDeploy == "enable" { //nolint: goconst
+	if dagDeploy == enable { //nolint: goconst
 		dagDeployEnabled = true
 	} else {
 		dagDeployEnabled = false
 	}
-
-	createInput := &astro.CreateDeploymentInput{
-		WorkspaceID:           workspaceID,
-		ClusterID:             clusterID,
-		Label:                 label,
-		Description:           description,
-		DagDeployEnabled:      dagDeployEnabled,
-		RuntimeReleaseVersion: runtimeVersion,
-		DeploymentSpec:        spec,
-		APIKeyOnlyDeployments: *enforceCD,
+	createDeploymentRequest := astroplatformcore.CreateDeploymentRequest{}
+	// build standard input
+	if IsDeploymentStandard(deploymentType) || IsDeploymentDedicated(deploymentType) {
+		var highAvailabilityValue bool
+		if highAvailability == enable { //nolint: goconst
+			highAvailabilityValue = true
+		} else {
+			highAvailabilityValue = false
+		}
+		var workerConcurrency int
+		for i := range configOption.WorkerMachines {
+			if strings.EqualFold(configOption.DefaultValues.WorkerMachineName, configOption.WorkerMachines[i].Name) {
+				workerConcurrency = int(configOption.WorkerMachines[i].Concurrency.Default)
+			}
+		}
+		defautWorkerQueue := []astroplatformcore.WorkerQueueRequest{{
+			AstroMachine:      astroplatformcore.WorkerQueueRequestAstroMachine(configOption.DefaultValues.WorkerMachineName),
+			IsDefault:         true,
+			MaxWorkerCount:    int(configOption.WorkerQueues.MaxWorkers.Default),
+			MinWorkerCount:    int(configOption.WorkerQueues.MinWorkers.Default),
+			Name:              "default",
+			WorkerConcurrency: workerConcurrency,
+		}}
+		if defaultTaskPodCpu == "" {
+			defaultTaskPodCpu = configOption.ResourceQuotas.DefaultPodSize.Cpu.Default
+		}
+		if defaultTaskPodMemory == "" {
+			defaultTaskPodMemory = configOption.ResourceQuotas.DefaultPodSize.Memory.Default
+		}
+		if resourceQuotaCpu == "" {
+			resourceQuotaCpu = configOption.ResourceQuotas.ResourceQuota.Cpu.Default
+		}
+		if resourceQuotaMemory == "" {
+			resourceQuotaMemory = configOption.ResourceQuotas.ResourceQuota.Memory.Default
+		}
+		if IsDeploymentStandard(deploymentType) {
+			var requestedCloudProvider astroplatformcore.CreateStandardDeploymentRequestCloudProvider
+			if cloudProvider == gcpCloud {
+				requestedCloudProvider = astroplatformcore.CreateStandardDeploymentRequestCloudProviderGCP
+			}
+			if cloudProvider == awsCloud {
+				requestedCloudProvider = astroplatformcore.CreateStandardDeploymentRequestCloudProviderAWS
+			}
+			if cloudProvider == azureCloud {
+				requestedCloudProvider = astroplatformcore.CreateStandardDeploymentRequestCloudProviderAZURE
+			}
```

Comment:
```suggestion
			if cloudProvider == gcpCloud {
				requestedCloudProvider = astroplatformcore.CreateStandardDeploymentRequestCloudProviderGCP
			}
			else if cloudProvider == awsCloud {
				requestedCloudProvider = astroplatformcore.CreateStandardDeploymentRequestCloudProviderAWS
			}
			else if cloudProvider == azureCloud {
				requestedCloudProvider = astroplatformcore.CreateStandardDeploymentRequestCloudProviderAZURE
			}
```

Reply:
the linter asks you to make it switch if you do this so I'll make it a switch